lib/src/fileset.rs

```diff
@@ -227,18 +228,6 @@ impl FilesetExpression {
         FilesetExpression::Pattern(FilePattern::PrefixPath(path))
     }
 
-    /// Expression that matches either `self` or `other` (or both).
-    pub fn union(self, other: Self) -> Self {
-        match self {
-            // Micro optimization for "x | y | z"
-            FilesetExpression::UnionAll(mut expressions) => {
-                expressions.push(other);
-                FilesetExpression::UnionAll(expressions)
-            }
-            expr => FilesetExpression::UnionAll(vec![expr, other]),
-        }
-    }
-
```

Comment:
It might be nice to keep this helper function around so that you can still `files1.union(files2)`?

Reply:
We can reintroduce a simpler version if needed?

I personally prefer `union_all([x, y])` than method chain `x.union(y)` as `x` can also be a multi-line chained expression.

Reply:
Seems reasonable; I was thinking that `.union()` is a closer match for the infix syntax used in the expression language, but the only uses in my `.mailmap` changes are actually prefix `RevsetExpression::union(a, b)`, so they can be migrated over without any stylistic change.

What do you think about changing `union_all` to take `impl Into<Vec<FilesetExpression>>` so you can save some syntactic noise by writing `RevsetExpression::union_all([a, b])` rather than `RevsetExpression::union_all(vec![a, b])`?

Reply:
Oh, I just noticed that this doesn’t remove or change `RevsetExpression::union` at all – is that intentional?

Reply:
> Oh, I just noticed that this doesn’t remove or change `RevsetExpression::union` at all – is that intentional?

Yes. `RevsetExpression::Union` isn't migrated to `Vec<RevsetExpression>` yet, so the underlying representation is still binary form.

Reply:
Makes sense, sorry for the noise then!